src/lib/token.ts

```diff
@@ -0,0 +1,89 @@
+import { Config, TokenSearchRules, TokenOptions } from '../types'
+import { encode64 } from './utils'
+import crypto from 'crypto'
+
+/**
+ * Create the header of the token.
+ *
+ * @param {String} apiKey API key used to sign the token.
+ * @param {String} encodedHeader Header of the token in base64.
+ * @param {String} encodedPayload Payload of the token in base64.
+ * @returns {String} The signature of the token in base64.
+ */
+function sign(apiKey: string, encodedHeader: string, encodedPayload: string) {
+  return crypto
+    .createHmac('sha256', apiKey)
+    .update(`${encodedHeader}.${encodedPayload}`)
+    .digest('base64')
+    .replace(/\+/g, '-')
+    .replace(/\//g, '_')
+    .replace(/=/g, '')
+}
+
+/**
+ * Create the header of the token.
+ *
+ * @returns {String} The header encoded in base64.
+ */
+function createHeader() {
+  const header = {
+    alg: 'HS256',
+    typ: 'JWT',
+  }
+
+  return encode64(header).replace(/=/g, '')
+}
+
+/**
+ * Create the payload of the token.
+ *
+ * @param {SearchRules} searchRules Search rules that are applied ton every search.
+ * @param {String} apiKey Api key used as issuer of the token.
+ * @param {Date | undefined} apiKey Date value.
```

Comment:
I think is the expiration date of the token not of the API key is could be confusing